test/provider_test.exs

```diff
@@ -48,7 +48,11 @@ defmodule ProviderTest do
       System.put_env("OPT_6", "false")
       System.put_env("OPT_7", "3.14")
 
+      {:ok, pid} = TestModule.start_link()
+
       assert TestModule.load!() == :ok
+
+      GenServer.stop(pid)
```

Comment:
Also, this is a bit unreliable, because if the assertion fail, this won't execute. Now, the server will still be stopped, but it happens asynchronously, so it might happen while the next test is running. And because all of the servers use the same registered name, the next test might fail.